packages/snaps-controllers/src/snaps/location/http.ts

```diff
@@ -56,9 +56,11 @@ export class HttpLocation implements SnapLocation {
       this.url,
     ).toString();
 
-    const contents = await (
-      await this.fetchFn(canonicalPath, this.fetchOptions)
-    ).text();
+    const response = await this.fetchFn(canonicalPath, this.fetchOptions);
+    if (!response.ok) {
+      throw new Error(`Failed to fetch "${canonicalPath}".`);
```

Comment:
Could be helpful to include a status code here:
```suggestion
      throw new Error(`Failed to fetch "${canonicalPath}". Status code: ${response.status}.`);
```